code/modules/power/cable.dm

```diff
@@ -415,3 +412,199 @@
+		var/disy = normal_cable.y - src.y
+		if (disy == 1)
+			if (disx == 1)
+				if (normal_cable.d1 == SOUTHWEST || normal_cable.d2 == SOUTHWEST)
+					cable_surr |= NORTHEAST_UNIQUE
+			else if (disx == -1)
+				if (normal_cable.d1 == SOUTHEAST || normal_cable.d2 == SOUTHEAST)
+					cable_surr |= NORTHWEST_UNIQUE
+			else if (normal_cable.d1 == SOUTH || normal_cable.d2 == SOUTH)
+				cable_surr |= NORTH
+		else if (disy == -1)
+			if (disx == 1)
+				if (normal_cable.d1 == NORTHWEST || normal_cable.d2 == NORTHWEST)
+					cable_surr |= SOUTHEAST_UNIQUE
+			else if (disx == -1)
+				if (normal_cable.d1 == NORTHEAST || normal_cable.d2 == NORTHEAST)
+					cable_surr |= SOUTHWEST_UNIQUE
+			else if (normal_cable.d1 == NORTH || normal_cable.d2 == NORTH)
+				cable_surr |= SOUTH
+		else if (disy == 0)
+			if (disx == 1)
+				if (normal_cable.d1 == WEST || normal_cable.d2 == WEST)
+					cable_surr |= EAST
+			else if (disx == -1)
+				if (normal_cable.d1 == EAST || normal_cable.d2 == EAST)
+					cable_surr |= WEST
+		// the 'real' wires override and always connect to prevent loose ends
+		// cable_surr is any direction that needs to be connected to at all
+		// this bit does not get optimised
+/// causes cablespawner to spawn cables (amazing)
+/obj/cablespawner/proc/replace()
+	var/list/directions = list()
+	if (cable_surr & NORTH)
+		directions += NORTH
+	if (cable_surr & SOUTH)
+		directions += SOUTH
+	if (cable_surr & EAST)
+		directions += EAST
+	if (cable_surr & NORTHEAST_UNIQUE)
+		directions += NORTHEAST
+	if (cable_surr & SOUTHEAST_UNIQUE)
+		directions += SOUTHEAST
+	if (cable_surr & WEST)
+		directions += WEST
+	if (cable_surr & NORTHWEST_UNIQUE)
+		directions += NORTHWEST
+	if (cable_surr & SOUTHWEST_UNIQUE)
+		directions += SOUTHWEST
+
+	if (length(directions) == 0)
+		cable_laying(0,NORTH)
+	// multiple cables, spiral out from the centre
+	else if (src.override_centre_connection)
+		for (var/i in 1 to length(directions))
+			cable_laying(0, directions[i])
+	else if (length(directions) >= 3)
+		for (var/i in 1 to length(directions))
+			cable_laying(0, directions[i])
+	else if (length(directions) == 1)
+	// end of a cable
+		cable_laying(0, directions[1])
+	else if (length(directions) == 2)
+	// a normal, single cable
+		cable_laying(directions[1], directions[2])
+	qdel(src)
+
+/// places a cable with d1 and d2
+/obj/cablespawner/proc/cable_laying(var/dir1, var/dir2)
+	var/obj/cable/current = new src.cable_type(src.loc)
+	current.d1 = dir1
+	current.d2 = dir2
+	current.color = src.color
+	current.UpdateIcon()
+
+/obj/cablespawner/black
+	color = "#272e30"
+/obj/cablespawner/blue
+	color = "#009ae7"
+/obj/cablespawner/brown
+	color = "#633221"
+/obj/cablespawner/green
+	color = "#03c53d"
+/obj/cablespawner/hotpink
+	color = "#FF69B4"
+/obj/cablespawner/orange
+	color = "#FF5D06"
+/obj/cablespawner/purple
+	color = "#9b59b6"
+/obj/cablespawner/white
+	color = "#ffffff"
+/obj/cablespawner/yellow
+	color = "#fffb00"
```

Comment:
These should spawn the appropriate color type (macro-defined in `cablecolors.dm`). I.e. `/obj/cable/yellow`.
```suggestion
/obj/cablespawner/black
	color = "#272e30"
	cable_type = /obj/cable/black
/obj/cablespawner/blue
	color = "#009ae7"
	cable_type = /obj/cable/blue
/obj/cablespawner/brown
	color = "#633221"
	cable_type = /obj/cable/brown
/obj/cablespawner/green
	color = "#03c53d"
	cable_type = /obj/cable/green
/obj/cablespawner/hotpink
	color = "#FF69B4"
	cable_type = /obj/cable/hotpink
/obj/cablespawner/orange
	color = "#FF5D06"
	cable_type = /obj/cable/orange
/obj/cablespawner/purple
	color = "#9b59b6"
	cable_type = /obj/cable/purple
/obj/cablespawner/white
	color = "#ffffff"
	cable_type = /obj/cable/white
/obj/cablespawner/yellow
	color = "#fffb00"
	cable_type = /obj/cable/yellow
```

Reply:
Note that it might be a much better idea to do this in the `cablecolors.dm` macro instead of doing it manually, That way if new cable colors are added this won't need a manualy update. Ping me in Discord if you need help with that.